src/ucp/rma/amo_send.c

```diff
@@ -205,18 +205,13 @@ UCS_PROFILE_FUNC(ucs_status_ptr_t, ucp_atomic_op_nbx,
         ucp_amo_init_fetch(req, ep, param->reply_buffer,
                            ucp_uct_atomic_op_table[opcode], op_size,
                            remote_addr, rkey, value, rkey->cache.amo_proto);
-        status_p = ucp_rma_send_request(req, param);
     } else {
         ucp_amo_init_post(req, ep, ucp_uct_atomic_op_table[opcode], op_size,
                           remote_addr, rkey, value, rkey->cache.amo_proto);
-
-        status_p = ucp_rma_send_request(req, param);
-        if (UCS_PTR_IS_PTR(status_p)) {
-            ucp_request_release(status_p);
-        }
-        status_p = UCS_STATUS_PTR(UCS_OK);
```

Comment:
this seems to break UCX atomic mca in OMPI (see `mca_atomic_ucx_op`)
in the ends it checks status by `ucx_status_to_oshmem`, which expects `UCS_OK` only

Reply:
right. but i think mca_atomic_ucx_op() is wrong here: since according to API design atomic_nbx is allowed to return a request
so need to fix OMPI code

Reply:
the current implementation does not break API semantic, it just always returns `UCS_OK`, which means cb will not be called. We implemented it this way due to performance reasons (after long discussions), the code in OMPI is bound to this behavior

Reply:
@gleon99

Reply:
do you remember what was the performance reason?
the reason to return a request now is to create a backpressure to the sender, to avoid excess memory allocation from requests mpool.

Reply:
shmem atomic benchmarks, we preserved the behavior of post operation as it is fast local operation not depending on the progress

Reply:
> do you remember what was the performance reason?
> the reason to return a request now is to create a backpressure to the sender, to avoid excess memory allocation from requests mpool.

but we release taken request immediately in atopic function, why does it cause mempool growth?

Reply:
> > do you remember what was the performance reason?
> > the reason to return a request now is to create a backpressure to the sender, to avoid excess memory allocation from requests mpool.
> 
> but we release taken request immediately in atomic function, why does it cause mempool growth?

sorry, got it. it's alive until op completes

Reply:
@Artemy-Mellanox seems we cannot add the "backpressure" (e.g returning a request from atomic_nbx post) before making the corresponding change in OpenMPI mca/atomic/ucx
so can you pls move this change to next PR, and keep only perftest stuff in this PR? and in meantime, make the necessary changes in OpenMPI.